src/ops/fuelup_component/list.rs

```diff
@@ -0,0 +1,101 @@
+use crate::{
+    commands::component::ListCommand, download::get_latest_version, fmt::bold, toolchain::Toolchain,
+};
+use anyhow::Result;
+use component::Components;
+use semver::Version;
+use std::io::Write;
+use tracing::info;
+
+fn format_installed_component_info(
+    name: &str,
+    version: Option<String>,
+    version_info: &str,
+) -> String {
+    if let Some(v) = version {
+        format!("  {name} {v} ({version_info})\n")
+    } else {
+        format!("  {name} : failed getting current version\n")
+    }
+}
+
+fn format_installable_component_info(name: &str, latest_version: &str) -> String {
+    format!("  {name} (latest: {latest_version})\n")
+}
+
+fn format_forc_default_plugins(plugin_executables: Vec<String>) -> String {
+    format!(
+        "    - {}\n",
+        plugin_executables
+            .iter()
+            .filter(|c| *c != component::FORC)
+            .map(|s| format!("{s} "))
+            .collect::<String>(),
+    )
+}
+
+pub fn list(_command: ListCommand) -> Result<()> {
+    let toolchain = Toolchain::from_settings()?;
+
+    // use write! instead of writeln! here to prevent this from printing first.
+    bold(|s| write!(s, "{}", toolchain.name));
+
+    let mut installed_components_summary = String::from("\nInstalled:\n");
+    let mut available_components_summary = String::from("Installable:\n");
+
+    let components = Components::collect_publishables()?;
+    for component in components {
+        let latest_version = get_latest_version(&component.name).map_or_else(
+            |_| String::from("failed to get latest version"),
+            |v| v.to_string(),
+        );
+        if toolchain.has_component(&component.name) {
+            let exec_path = toolchain.bin_path.join(&component.name);
+
+            let current_version = if let Ok(o) = std::process::Command::new(exec_path)
+                .arg("--version")
+                .output()
+            {
+                let output = String::from_utf8_lossy(&o.stdout).into_owned();
+                output.split_whitespace().nth(1).map_or_else(
+                    || None,
+                    |v| Version::parse(v).map_or_else(|_| None, |v| Some(v.to_string())),
+                )
+            } else {
+                None
+            };
+
+            let version_info = match Some(&latest_version) == current_version.as_ref() {
+                true => "up-to-date".to_string(),
+                false => format!("latest: {}", &latest_version),
+            };
+
+            installed_components_summary.push_str(&format_installed_component_info(
+                &component.name,
+                current_version,
+                &version_info,
+            ));
+
+            if component.name == component::FORC {
+                installed_components_summary
+                    .push_str(&format_forc_default_plugins(component.executables))
```

Comment:
Are we sure we want to have a special case for the default plugins rather than listing them like the rest? Is it possible for a user to have a different version of `forc` to say `forc-lsp` if they're using a custom toolchain?

Reply:
Hmm, this is tricky because of the way we release forc binaries - they're all packaged together in one tarball in [this step](https://github.com/FuelLabs/sway/blob/12aff7873345daf8930affa472c222ffb7906c5d/.github/workflows/ci.yml#L626-L629). So it's not possible to install `forc-lsp` separately from `forc`, and listing them in the way I did was to emphasize that it's impossible to do that. Otherwise, we might have people trying to do `fuelup component add forc-lsp` when the command should be `fuelup component add forc`.

It's hard to get a different version of `forc` vs `forc-lsp`, but i guess its possible if the user manually moves their binaries around.

To address this, we could update this behaviour to list the versions of each plugin tagged to `forc` as well, similar to how `fuelup show` does it. But perhaps a longer term solution might be one of the below, to be able to install, say `forc-lsp`, separately:

1) we re-work the way we package the `forc` binaries (I've been wondering recently why we didn't do what [`tracing` does with their releases](https://github.com/tokio-rs/tracing/releases)), or
2)  re-work unpacking the tarball to keep only the binary we want to specifically install and delete the others (though this would probably over-complicate the process for little to no gain, so I don't really like this)

Reply:
> So it's not possible to install forc-lsp separately from forc, and listing them in the way I did was to emphasize that it's impossible to do that. Otherwise, we might have people trying to do fuelup component add forc-lsp when the command should be fuelup component add forc.

Ahhh if that's the case, let's not worry about it - this seems fine :+1: It would be a very niche need anyway, and there are trivial workarounds (like installing forc-lsp via cargo instead). We can reconsider splitting up `forc` if we get a motivating issue, but otherwise this seems fine.

Reply:
I think I will create a separate issue to print the versions of the plugins themselves as well, because it is indeed an oversight if the actual executables are moved around by the user and are different from the `forc` version.